pkg/tenant/topics.go

```diff
@@ -443,27 +439,20 @@ func (s *Server) TopicUpdate(c *gin.Context) {
 			return
 		}
 
-		// Create the Ensign context with the one-time claims
-		ensignContext := qd.ContextWithToken(ctx, rep.AccessToken)
-
 		// Archive means a "soft" delete in Ensign (no data is destroyed)
-		updateRequest := &pb.TopicMod{
-			Id:        t.ID.String(),
-			Operation: pb.TopicMod_ARCHIVE,
-		}
-		var tombstone *pb.TopicTombstone
-		if tombstone, err = s.ensign.DeleteTopic(ensignContext, updateRequest); err != nil {
-			if status.Code(err) == codes.NotFound {
-				sentry.Warn(c).Err(err).Str("topicID", updateRequest.Id).Msg("topic not found in ensign even though it is in tenant")
-				c.JSON(http.StatusNotFound, api.ErrorResponse("topic not found"))
+		if err = s.ensign.InvokeOnce(rep.AccessToken).ArchiveTopic(ctx, t.ID.String()); err != nil {
+			switch {
+			case err.Error() == "not implemented yet":
+				sentry.Warn(c).Err(err).Msg("this version of the Go SDK does not support topic archiving")
+				c.JSON(http.StatusNotImplemented, api.ErrorResponse("archiving a topic is not supported"))
```

Comment:
The Go SDK doesn't currently support topic delete or topic archive so this is a temporary workaround for the tests. It would also be nice to have standardized errors from the SDK, not sure if there's a story for that yet.

Reply:
There is - just a matter of getting to it!